### PR TITLE
Improve ACL results

### PR DESCRIPTION
**Pull request by hwine**
Two major changes:
- reduce false positives by requiring the match be as-a-token (word)
  - short strings often matched base64 encoded text
- remove false negatives when patterns were skipped after encountering a search limit

### CODE REVIEW

File: notebooks/UserSearchPy3.ipynb [hunk 1]
```diff
@@ -1328,6 +1331,41 @@
     "lines_to_next_cell": 1
    },
    "outputs": [],
+   "source": [
+    "def prune_hits_to_ignore(full_list, id_to_find):\n",
```
Comment (hwine) [inline notebooks/UserSearchPy3.ipynb:1335]: This code is what reduces false positives. Portions were present in the following cell, and refactored into this function.

File: notebooks/UserSearchPy3.ipynb [hunk 2]
```diff
@@ -1328,6 +1331,41 @@
...
+    "    for index, hit in enumerate(hit_list_1):\n",
+    "#         print(f\"  Hit {index} has {len(hit.text_matches)} contexts\")\n",
+    "        for ctxt, context in enumerate(hit.text_matches):\n",
+    "            if id_re.search(context[\"fragment\"]):\n",
+    "                hit_list_2.append(hit)\n",
+    "#                 print(f\"Adding hit {index}; context {ctxt} ({len(hit_list_2)=}): {context['fragment']}\")\n",
+    "                break\n",
+    "            else:\n",
+    "#                 print(f\"ignoring context {context['fragment']}\")\n",
+    "                ...\n",
```
Comment (hwine) [inline notebooks/UserSearchPy3.ipynb:1354]: This is the new code, which scans the fragments of each "hit" and only keeps those where the search term is a word.

File: notebooks/UserSearchPy3.ipynb [hunk 3]
```diff
@@ -1360,57 +1398,62 @@
...
-    "                        if gh:\n",
-    "                            # re-established connection\n",
-    "                            print(f\"re-established connection on try {i+1}\")\n",
-    "                            break\n",
+    "            assume_time_out = True\n",
+    "            while assume_time_out:\n",
+    "                try:\n",
+    "                    # 2023-05-25 can't use regex in code search, so return context for further processing\n",
+    "                    full_list = list(gh.search_code(query=f\"org:{org} {l}\", text_match=True))\n",
+    "                    assume_time_out = False\n",
```
Comment (hwine) [inline notebooks/UserSearchPy3.ipynb:1406]: This is where we recover the false negatives. We won't advance to the next token until we've had a successful search with the current one. This required increasing indentation, which makes for a "huge diff" - the key lines are the ones referencing `asssume_time_out`.

**Review by cknowles-moz — APPROVED** [review]
The explanatory text did wonders - particularly the bit explaining the huge diff due to indentation changes.  
Looks probable that it will do what it says on the tin.  Look forward to putting it through the paces.

**Pull request merged**